CONTRIBUTING.md

```diff
@@ -11,56 +11,117 @@ Table of Contents:
+Code should be documented where appropriate. The existing code can be used as a guidance here and
+the general `rustfmt` rules can be followed for formatting.
 
-If any change has been made to the `config.rs` file, these changes should also be documented in the example configuration file `alacritty.yml`.
+If any change has been made to the `config.rs` file, these changes should also be documented in the
+example configuration file `alacritty.yml`.
 
-Changes compared to the latest Alacritty release which have a direct effect on the user (opposed to things like code refactorings or documentation/tests) additionally need to be documented in the `CHANGELOG.md`. The existing entries should be used as a style guideline. The change log should be used to document changes from a user-perspective, instead of explaining the technical background (like commit messages). More information about Alacritty's change log format can be found [here](https://keepachangelog.com).
+Changes compared to the latest Alacritty release which have a direct effect on the user (opposed to
+things like code refactorings or documentation/tests) additionally need to be documented in the
+`CHANGELOG.md`. The existing entries should be used as a style guideline. The change log should be
+used to document changes from a user-perspective, instead of explaining the technical background
+(like commit messages). More information about Alacritty's change log format can be found
+[here](https://keepachangelog.com).
 
 ### Style
 
-All Alacritty changes are automatically verified by CI to conform to its rustfmt guidelines. If a CI build is failing because of formatting issues, you can install rustfmt using `rustup component add rustfmt` and then format all code using `cargo fmt`.
+All Alacritty changes are automatically verified by CI to conform to its rustfmt guidelines. If a CI
+build is failing because of formatting issues, you can install rustfmt using `rustup component add
+rustfmt` and then format all code using `cargo fmt`.
+
+# Release Process
+
+Alacritty's release process aims to provide stable and well tested releases without having to hold
+back new features during the testing period.
+
+To achieve these goals, a new branch is created for every new release. Both the release candidates
+and the final version are only comitted and tagged in this branch. The master branch only tracks
+development versions, allowing us to keep the branches completely separate without merging releases
+back into master.
+
+The exact steps for an exemplary `1.2.3` might look like this:
+ 1. Initially, the version on the latest master is `1.2.3-dev`
+ 2. A new `v1.2.3` branch is created for the release
+ 3. On master, the version is bumped to `1.2.4-dev`
+        and the `-dev` is stripped from previous change log entries
+ 4. In the branch, the version is bumped to `1.2.3-rc1`
```

Comment:
> Should we enumerate the places the version string appears here, or just hint at searching for the old string? Is this obvious?

I think having a list would be kinda troublesome, because if a list exists, people expect that list to be correct. I can totally see us forgetting to add a new file to the list and then we forget to bump it.

The best way to check is probably to just run `git show vX.Y.Z --stat` and that will list all files bumped in the last release, that's probably the best way to go about it and that's what I do. Doing a `grep` in the repo if you're uncertain should leave you with way more certainty than a centralized list.

> Also, maybe we should indicate that the versions should be 1.2.3-rcX since this step is referred to in a loop.

The thing with this is that we never print a full example and I think examples is exactly what you want to orient yourself with. I would expect someone who makes releases to realize that the rc1 should be bumped to rc2, given our previous RC periods, however if someone has no exact reference of what such a tag might look like, it might be more confusing (start with 0 or 1 for example).

I've thought about spelling it out twice, with the second part being repeated and the rc1 being layed out on its own, but I think that would just be too much. I think we can expect at least a little bit of thinking.